src/plugins/ucx/ucx_backend.cpp

```diff
@@ -427,11 +426,44 @@ class nixlUcxBackendH : public nixlBackendReqH {
  * Progress thread management
 *****************************************/
 
-void nixlUcxEngine::progressFunc()
+nixlUcxThreadEngine::nixlUcxThreadEngine(const nixlBackendInitParams &init_params)
+    : nixlUcxEngine(init_params) {
+    if (!nixlUcxMtLevelIsSupported(nixl_ucx_mt_t::WORKER)) {
+        NIXL_ERROR << "UCX library does not support multi-threading";
+        this->initErr = true;
```

Comment:
Please take into account this PR https://github.com/ai-dynamo/nixl/pull/537.
We try to get rid of using of `initErr`, and throw/catch exceptions instead.